tests/src/rules/extensions.js

```diff
@@ -33,6 +33,16 @@ ruleTester.run('extensions', rule, {
       settings: { 'import/resolve': { 'extensions': [ '.js', '.jsx', '.json' ] } },
     }),
 
+    test({
+      code: [
+        'import lib from "./bar"',
+        'import lib from "./bar.json"',
+        'import lib from "./bar.hbs"',
+      ].join('\n'),
+      options: [ 'always', { js: 'never', jsx: 'never' } ],
+      settings: { 'import/resolve': { 'extensions': [ '.js', '.jsx', '.json', '.hbs' ] } },
+    }),
+
```

Comment:
Maybe add a test case with never, for completeness' sake.
I'd also like to see both valid and invalid test cases with this change, if it's not too much to ask.